### DISCUSSION TITLE
How to get openssl to succesfully compile in guix or gitian build

### DISCUSSION THREAD

**Discussion by gdiscord** [openssl/openssl#23645] [Q&A]
I'm building a bitcoin fork in guix, but openssl reports error and doesn't successfully compile
This is the error:

  636 |     sqr_add_c2(a, 3, 2, c3, c1, c2);
      |     ^~~~~~~~~~
crypto/bn/asm/x86_64-gcc.c:365:17: error: expected ')' before ':' token
  365 |                 : "=a"(t1),"=d"(t2)     \
      |                 ^
crypto/bn/asm/x86_64-gcc.c:639:5: note: in expansion of macro 'sqr_add_c'
  639 |     sqr_add_c(a, 3, c1, c2, c3);
      |     ^~~~~~~~~
crypto/bn/asm/x86_64-gcc.c:369:17: error: expected ')' before ':' token
  369 |                 : "+r"(c0),"+r"(c1),"+r"(c2)            \
      |                 ^
crypto/bn/asm/x86_64-gcc.c:639:5: note: in expansion of macro 'sqr_add_c'
  639 |     sqr_add_c(a, 3, c1, c2, c3);

Not sure why that is happening because it compiles successfully directly on my computer.

Could this be a bug, and how can I resolve it so that it works?

**Comment by gdiscord** [comment]
My sincere apologies for the duplicate.

Went away and spent a couple of days researching to find an answer to this issue. I had suspected something to do with the compiler not correctly understanding asm macros. Towing that line my research effort landed me on the openssl wiki page, which revealed that there is a "no-asm" option that can be used for debugging purposes, but must not be used in production. Hence I came back here to search for a "proper solution", and it turns out someone else already raised this issue just a few weeks ago.
https://github.com/openssl/openssl/discussions/23382

This is the same issue I'm experiencing, albeit the built system is different. My build is using the dependency management of bitcoin repo to add openssl 3x.

https://github.com/bitcoin/bitcoin/blob/v0.19.2/depends/packages/openssl.mk

How do I pass the asm option to this?

thanks.

**Comment by gdiscord** [comment]
Should I close this one also and instead post in the case opened by the different user?

**Comment by levitte** [comment]
I'd say that this discussion is easier to find for those who struggle with guix / gitian, so let's keep it.  Since it refers to the "duplicate" issue, that's not lost.

I suggest you experiment with what's being said #23382 and tell us what you find, here.

**Comment by gdiscord** [comment]
Problem solved with -std=gnu11.

thanks
